flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1974,8 +2083,8 @@ def create(self):
 class MountBlockDeviceTests(
     make_istatechange_tests(
         MountBlockDevice,
-        dict(volume=_ARBITRARY_VOLUME, mountpoint=FilePath(b"/foo")),
-        dict(volume=_ARBITRARY_VOLUME, mountpoint=FilePath(b"/bar")),
+        dict(dataset_id=uuid4(), mountpoint=FilePath(b"/foo")),
```

Comment:
And this one is even more confusing to read because we're supplying the two dictionaries for the benefit of a `make_comparison_tests` whose test case is mixed in to the test case produced by `make_istatechange_tests`.
But even keyword args won't help in this case.